toml.md

```diff
@@ -36,25 +36,25 @@ should be easy to parse into data structures in a wide variety of languages.
 
 ## Spec
 
+A TOML file must be a valid UTF-8 encoded Unicode document.
+
 - TOML is case-sensitive.
-- A TOML file must be a valid UTF-8 encoded Unicode document.
-- Whitespace means tab (0x09) or space (0x20).
-- Newline means LF (0x0A) or CRLF (0x0D 0x0A).
+- Whitespace means tab (U+0009) or space (U+0020).
+- Newline means LF (U+000A) or CRLF (U+000D U+000A).
 
 ## Comment
 
 A hash symbol marks the rest of the line as a comment, except when inside a
-string.
+string. Comments may contain any valid code points except a limited subset of
```

Comment:
I would change "valid code points" to "characters". Since we already state that that the whole document must be valid UTF-8, "valid" seems an unnecessary duplication. Any elsewhere in the document we consistently talk about "characters" rather than "code points", so it seem a bit strange to deviate here from the usual terminology.

Reply:
There’s, though, one problem with “character”. A single character can consist of multiple code points, and Unicode says nothing about what a valid character is. You can, for instance, make your own by using several combining diacritics.

Furthermore, while we require valid UTF-8 byte streams, it’s possible to use invalid combinations of code points, like a smiley variant selector in the wrong position. Or a BOM in the middle of the file.

Reply:
OK, let's stick with "code points" then. But is there any good reason for the "valid"?

Reply:
I think it is in reference to the fact that, as I showed above, you _could_ encode invalid code points (i.e., code points above `0x10FFFF` for instance), or you _could_ consider encoding surrogate code points, but these aren't _invalid code points_, they are just _invalid code units_ in a UTF-8 byte stream.

Terminology is tricky. By the time you have _valid UTF-8_, you cannot have invalid code points (unless "invalid" means in the context of this specification). Perhaps this would be better?

> Comments may contain any Unicode code points except a limited subset of

Reply:
For as long as I've worked with Unicode, I've wondered what an intuitive replacement for the word "character" that wouldn't sound too alien to non-technical people could be. "Code point" is a bit formal, but to the extent that people need to be knowledgeable about these things, they need to know that code points are more fundamental than characters.

That said, I've used the final sentence above. Thanks for your help and insight @ChristianSi @abelbraaksma !